libtiny_client/src/state.rs

```diff
@@ -232,6 +271,31 @@ impl StateInner {
                 }
             }
 
+            // Reply 477 when user needs to be identified with NickServ to join a channel
+            Reply { num: 477, params } => {
+                // Only try to automatically rejoin if nickserv_ident is configured
+                if self.nickserv_ident.is_some() {
+                    // Get channel name
+                    let channel = params.get(1).unwrap().clone();
+                    // Find channel in self.chans
+                    let idx = utils::find_idx(&self.chans, |c| c.name == channel).unwrap();
+                    // Retry joining channel if retries are available
+                    if self.chans[idx].retry_join() {
+                        let mut snd_irc_msg = snd_irc_msg.clone();
```

Comment:
I hate having to clone here...I don't think there is any other way, right?

Reply:
This is how you do it, yes. It shouldn't be a problem (it should be cheap) as this type is supposed to be cloned and shared with threads/tasks.